pykafka/producer.py

```diff
@@ -395,9 +405,26 @@ def queue_reader():
                     # surface all exceptions to the main thread
                     self.producer._worker_exception = sys.exc_info()
                     break
+            try:
```

Comment:
This way just seems cleaner to me, since it doesn't involve freezing the entire producer to do a leader reassignment.